### PR TITLE
feat(trading): liquidation notification toast

### LINKED ISSUE #6055: Add a toast for liquidation notification
## The Chore

Add a toast when you have been liquidated.

Continuation of work in #6019 

## Tasks

- [x] Add toast
- [x] Make sure the toast message contains all the nice info
- [x] Make sure the toast pops up if the liquidation happened while you were away
- [x] Probably the toast will need to pop up every time you open console and your last trade is still the network close out, even if you've seen it before ... unless we can keep in local storage ... but I doont know that we need to
- [x] Once you have made a new trade the toast should not appear again next time you open console
- [x] In this toast there could be a button which sets the new "show closed positions" button to on so that you can see the position that was closed

### PR DESCRIPTION
**Pull request by asiaznik**
# Related issues 🔗

Closes #6055

# Description ℹ️

* adds toast that notifies about a position liquidation

# Demo 📺

Screenshot 2024-06-20 at 11 14 02

https://github.com/vegaprotocol/frontend-monorepo/assets/1980305/a2c08b5b-d32d-4adf-9733-4700e70d97b2

# Technical 👨‍🔧

N/A

### CODE REVIEW

File: apps/trading/lib/hooks/use-position-closeout-notification.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      .filter((v) => !v.startsWith(remove))
+      .concat(add)
+  );
+};
+
+export const usePositionCloseoutNotification = () => {
+  const client = useApolloClient();
+  const { pubKey } = useVegaWallet();
+  const [errorCount, setErrorCount] = useState(0);
+  const onRejection = useCallback(
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-position-closeout-notification.ts:92]: Use of useCallback here is pointless

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/lib/hooks/use-position-closeout-notification.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+const LOCAL_STORAGE_KEY = 'vega_position_close_out_notification';
+
+const storedPushedNotification: string[] | undefined = undefined;
+
+const readStoredPushedNotification = () => {
+  if (storedPushedNotification) {
+    return storedPushedNotification;
+  }
+  let value: string[] = [];
+  const rawValue = LocalStorage.getItem(LOCAL_STORAGE_KEY);
```
[the remark this answers is not in the document]
Comment (mattrussell36) [reply apps/trading/lib/hooks/use-position-closeout-notification.ts:29]: On second thought, do we even need local storage? I'm struggling to see the benefit of using it

File: apps/trading/lib/hooks/use-position-closeout-notification.tsx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+  );
+
+  const highlightPositionElement = () => {
+    let tries = 0;
+    const int = setInterval(() => {
+      tries += 1;
+      if (tries > 10) {
+        clearInterval(int);
+      }
+      const position = document.querySelector(
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-position-closeout-notification.tsx:102]: Dont like this. If we move to a persisted zustand store we can access it in the positions table

File: apps/trading/lib/hooks/use-position-closeout-notification.tsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+      // hides toast instead of removing is so it won't be re-added on rerender
+      updateToast(createToastId(tradeId), { hidden: true });
+      setSeen(tradeId);
+    },
+    [setSeen, updateToast]
+  );
+
+  const [errorCount, setErrorCount] = useState(0);
+
+  const onRejection = useCallback(
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-position-closeout-notification.tsx:178]: Pointless useCallback

File: apps/trading/lib/hooks/use-position-closeout-notification.tsx [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+  useEffect(() => {
+    let subscription: { unsubscribe: () => void } | undefined = undefined;
+    let ignore = false;
+
+    if (!pubKeys) {
+      return;
+    }
+
+    const runQueries = () => {
+      if (ignore) return;
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-position-closeout-notification.tsx:265]: The ignore needs to be used inside the async callbacks to properly ignore 'stale' requests. This is another reason to just use the generated hook, which should handle race conditions if the fetch is invoked twice rapidly

File: apps/trading/lib/hooks/use-position-closeout-notification.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattrussell36) [reply apps/trading/lib/hooks/use-position-closeout-notification.ts]: You can disregard the above comment, it makes sense to me now

File: apps/trading/lib/hooks/use-position-closeout-notification.tsx [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+      trades: [],
+      addTrade: (tradeId) => {
+        set((state) => {
+          const trades = uniq([...state.trades, tradeId]);
+          return { trades };
+        });
+      },
+    }),
+    {
+      name: 'seen_close_out_positions_toasts',
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-position-closeout-notification.tsx:54]: `vega_` prefix

**Review by mattrussell36 — APPROVED** [review]

File: apps/trading/lib/hooks/use-position-closeout-notification.tsx (hunk 6 above)
Comment (asiaznik) [inline apps/trading/lib/hooks/use-position-closeout-notification.tsx:54]:
```suggestion
      name: 'vega_seen_close_out_positions_toasts',
```

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**